MACE_Frontend/src/components/map/index.tsx

```diff
@@ -123,13 +180,65 @@ export default class MapView extends React.Component<Props, State> {
         />
 
         <Markers />
-        
+
         <ContextMenu
           position={this.state.contextMenuPosition}
           visible={this.state.contextMenuVisible}
           onRequestClose={() => this.removeContextMenu()}
-          actions={[{ label: "Set global origin", action: this.setGlobalOrigin }]}
+          actions={[{ label: "Set global origin", action: this.setSwarmOrigin }, { label: "Set Vehicle origin", action: this.togglePopup }]}
         />
+        {this.state.popupVisible &&
+          <Popup
+            position={[this.state.originPosition.lat, this.state.originPosition.lng]}
+            autoClose={false}
+            closeOnClick={true}
+            closeButton={false}
+            offset={[10, 15]}
+          >
+            <span style={styles.inputLabel}>Select Aircraft:</span>
+            <div>
+              <select style={styles.selectRow} onChange={this.onAircraftSelected} >
```

Comment:
Did you try using a `<Select />` component from `react-select`? This could allow for multiple selection options if we wanted to send a subset of vehicles a new home location. I think it would look something like this: 

```
<Select
    options={this.createDropdownList}
    isMulti
    onChange={(ev) => {
        this.onAddSelectedAircraft(ev);
    }}
/>
```
where `onAddSelectedAircraft` would simply append the aircraft to a list. You would have to handle the removal as well.